portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/Prototype/MockConfiguration.jsx

```diff
@@ -0,0 +1,271 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+    Dialog,
+    DialogTitle,
+    DialogContent,
+    DialogActions,
+    TextField,
+    Button,
+    Select,
+    MenuItem,
+    FormControl,
+    InputLabel,
+} from '@mui/material';
+import { Alert } from 'AppComponents/Shared';
+
+const MockConfiguration = ({ open, onClose, configuration, setConfiguration, 
+    currentConfig, mockScripts, setMockScripts, paths, updatePaths, simulationSplitString }) => {
+    const [mockSimulation, setMockSimulation] = useState({});
+    const [currentEndpoint, setCurrentEndpoint] = useState(null);
+    const xMediationScriptProperty = 'x-mediation-script';
+
+    useEffect(() => {
+        if (open) {  // Only update when dialog opens
+            if (currentConfig.path) {
+                const {path, method} = currentConfig
+                setCurrentEndpoint({ path, method });
+                setMockSimulation(
+                    configuration?.config?.simulationDetails?.[path]?.[method] || {
+                        latency: 0,
+                        error: '0',
+                    }
+                );
+            } else {
+                setCurrentEndpoint(null);
+                setMockSimulation(
+                    configuration.config?.simulationDetails?.api || {
+                        latency: 0,
+                        error: '0',
+                    }
+                );
+            }
+        }
+    }, [currentConfig, configuration, open]);
+
+    const errorOptions = [
+        { value: '0', label: 'No Error' },
+        { value: '400', label: '400-Bad Request' },
+        { value: '401', label: '401-Unauthorized' },
+        { value: '403', label: '403-Forbidden' },
+        { value: '404', label: '404-Not Found' },
+        { value: '500', label: '500-Internal Server Error' },
+        { value: '501', label: '501-Not Implemented' },
+    ];
+
+    const handleChange = ({ target: { name, value } }) => {
+        if (name === 'latency' && !/^\d*$/.test(value)) return;
+        setMockSimulation(prev => ({ ...prev, [name]: value }));
+    };
+    
+    
+
+    const handleSave = () => {
+        const applySimForEndpoint = (path, method, simulation, apiSim = false) => {
```

Comment:
Would be better to use the entire word `Simulation` instead of using `Sim` for better understandability.